### ISSUE TITLE
[Bug]: /api/market/forecast Has No Authentication — Triggers Expensive LSTM Training on First Call

### ISSUE THREAD

**Issue by ameyvaidya44** [Eshajha19/agri#592]
**File**: main.py, lines 1206–1244

**Severity**: Level 3

**Description**:

The market price forecast endpoint has no authentication check:

```
@app.post("/api/market/forecast")
@limiter.limit("10/minute")
async def market_price_forecast(data: MarketForecastRequest, request: Request):
    try:
        from ml.price_forecaster import price_forecaster
        result = price_forecaster.forecast(
            commodity=data.commodity,
            days=data.days,
        )
        return result
```
The PriceForecaster._get_or_train() method trains a TensorFlow LSTM model on first call for each commodity and caches it in memory. Training runs for 30 epochs on the embedded dataset. The warm-up thread in init_ml_pipeline() pre-trains all 6 commodities at startup, but the commodity field in MarketForecastRequest accepts any string up to 60 characters — including values not in HISTORICAL_PRICES. When an unknown commodity is submitted, _get_or_train() falls back to Wheat data and trains a new model for that commodity name, caching it under the submitted string as the key.

An attacker can submit thousands of unique commodity strings (e.g., "Wheat1", "Wheat2", ...) each triggering a new 30-epoch LSTM training run. Each training run:

Consumes significant CPU for ~1–5 seconds
Allocates a TensorFlow model in memory (~several MB)
Is cached permanently in self._models with no eviction
The rate limit of 10/minute per IP is the only protection, but it applies per IP — an attacker with multiple IPs or a botnet can bypass it. More critically, the commodity field has no whitelist validation — it accepts any string, so the model cache grows without bound.

Compare this to /api/reports/generate which requires ["expert", "admin"] role, and /api/whatsapp/trigger-alert which we just fixed to require authentication. This endpoint, which triggers ML training and memory allocation, has no authentication at all.

**Impact**: Unauthenticated actors can trigger unbounded memory growth by submitting unique commodity strings, each causing a new LSTM model to be trained and cached. Sustained attack causes OOM crashes. CPU exhaustion during training degrades response times for all users. No authentication means competitors or scrapers can freely access price forecasting without being users of the platform.

**Comment by Eshajha19** [comment]
Assigned @ameyvaidya44
